src/widgets/panel/__tests__/panel.test.tsx

```diff
@@ -0,0 +1,399 @@
+/**
+ * @jest-environment jsdom
+ */
+/** @jsx h */
+import { h } from 'preact';
+
+import { createSearchClient } from '../../../../test/mock/createSearchClient';
+import instantsearch from '../../../index.es';
+import { wait } from '../../../../test/utils/wait';
+import panel from '../panel';
+import {
+  createMultiSearchResponse,
+  createSingleSearchResponse,
+} from '../../../../test/mock/createAPIResponse';
+import refinementList from '../../refinement-list/refinement-list';
+
+beforeEach(() => {
+  document.body.innerHTML = '';
+});
+
+describe('panel', () => {
+  describe('templates', () => {
+    test('renders with templates using `html`', async () => {
+      const container = document.createElement('div');
+      const searchClient = createMockedSearchClient();
+
+      const search = instantsearch({
+        indexName: 'indexName',
+        searchClient,
+      });
+
+      search.addWidgets([
+        panel({
+          templates: {
+            header({ results }, { html }) {
+              return html`<span
+                >Brand ${results && `(${results.nbHits} results)`}</span
+              >`;
+            },
+            footer({ results }, { html }) {
+              return html`<a href="#"
+                >More brands ${results && `(${results.nbHits} results)`}</a
+              >`;
+            },
+            collapseButtonText({ collapsed }, { html }) {
+              return html`<span>${collapsed ? 'More' : 'Less'}</span>`;
+            },
+          },
+          collapsed: () => true,
+        })(refinementList)({ container, attribute: 'brand' }),
```

Comment:
maybe this should be tested with a smaller widget so the snapshot is more focused?

Reply:
Yes this is better, I switched for `stats` in 8ec8259e.